is_writable_intern - build-break on ubuntu gcc 4.8.4

trying to compile with qrb -Uv && qrb -qmIv && qrb -Olocal -Nv && qrb -Oui -Nv, I'm getting this error from is_writable_intern():

libtool: compile:  g++ -DHAVE_CONFIG_H -I../include/qore -I../include -I../include -I../lib -DNDEBUG=1 -DHAVE_UNIX_CONFIG_H -D_QORE_LIB_INTERN -DMODULE_DIR=\"/home/kveton/qore/lib/x86_64-linux-gnu/qore-modules\" -DMODULE_VER_DIR=\"/home/kveton/qore/lib/x86_64-linux-gnu/qore-modules/0.8.12\" -DUSER_MODULE_DIR=\"/home/kveton/qore/share/qore-modules\" -DUSER_MODULE_VER_DIR=\"/home/kveton/qore/share/qore-modules/0.8.12\" -DHAVE_UNIX_CONFIG_H -g -O2 -m64 -fno-strict-aliasing -D_THREAD_SAFE -pthread -Wall -lm -g -O2 -m64 -fno-strict-aliasing -D_THREAD_SAFE -pthread -Wall -MT single-compilation-unit.lo -MD -MP -MF .deps/single-compilation-unit.Tpo -c single-compilation-unit.cpp  -fPIC -DPIC -o .libs/single-compilation-unit.o
In file included from /usr/include/fcntl.h:279:0,
                 from ../include/qore/QoreFile.h:38,
                 from ../include/qore/Qore.h:75,
                 from parser.ypp:32,
                 from single-compilation-unit.cpp:29:
In function 'int open(const char_, int, ...)',
    inlined from 'bool is_writable_intern(const QoreStringNode_)' at ql_file.cpp:81:53:
/usr/include/x86_64-linux-gnu/bits/fcntl2.h:50:26: error: call to '__open_missing_mode' declared with attribute error: open with O_CREAT in second argument needs 3 arguments
    __open_missing_mode ();
                          ^
In file included from ../include/qore/intern/QoreLibIntern.h:382:0,
                 from ../include/qore/Qore.h:222,
                 from parser.ypp:32,
                 from single-compilation-unit.cpp:29:
../include/qore/intern/qore_thread_intern.h: In member function 'void QoreFunction::parseInit()':
../include/qore/intern/qore_thread_intern.h:933:26: warning: 'pch.OptionalNamespaceParseContextHelper::ns' may be used uninitialized in this function [-Wmaybe-uninitialized]
          parse_set_ns(ns);
                          ^
In file included from single-compilation-unit.cpp:62:0:
Function.cpp:1694:40: note: 'pch.OptionalNamespaceParseContextHelper::ns' was declared here
    OptionalNamespaceParseContextHelper pch(ns);
                                        ^
In file included from single-compilation-unit.cpp:102:0:
QoreSocket.cpp: In member function 'BinaryNode\* QoreSocket::recvBinary(qore_offset_t, int, int_)':
QoreSocket.cpp:1276:18: warning: 'nrc' may be used uninitialized in this function [-Wmaybe-uninitialized]
    qore_offset_t nrc;
                  ^
QoreSocket.cpp: In member function 'BinaryNode_ QoreSocket::recvBinary(int, int_)':
QoreSocket.cpp:1284:18: warning: 'nrc' may be used uninitialized in this function [-Wmaybe-uninitialized]
    qore_offset_t nrc;
                  ^
QoreSocket.cpp: In member function 'QoreStringNode_ QoreSocket::recv(int, int_)':
QoreSocket.cpp:1314:18: warning: 'nrc' may be used uninitialized in this function [-Wmaybe-uninitialized]
    qore_offset_t nrc;
                  ^
QoreSocket.cpp: In member function 'QoreStringNode_ QoreSocket::recv(qore_offset_t, int, int_)':
QoreSocket.cpp:1306:18: warning: 'nrc' may be used uninitialized in this function [-Wmaybe-uninitialized]
    qore_offset_t nrc;
                  ^
make[4]: *_\* [single-compilation-unit.lo] Error 1
make[4]: Leaving directory `/home/kveton/src/qore/git/qore/lib'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory`/home/kveton/src/qore/git/qore/lib'
make[2]: **\* [all] Error 2
make[2]: Leaving directory `/home/kveton/src/qore/git/qore/lib'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory`/home/kveton/src/qore/git/qore'
make: **\* [all] Error 2

--- a/lib/ql_file.qpp
+++ b/lib/ql_file.qpp
@@ -78,7 +78,7 @@ static bool is_writable_intern(const QoreStringNode* path) {
 
    if (S_ISDIR(sbuf.st_mode)) { // If path is a directory.
       QoreStringMaker tfn("%s/xxx_qore_tmp_test_xxx", path->getBuffer());
-      rc = open(tfn.getBuffer(), O_WRONLY | O_CREAT);
+      rc = open(tfn.getBuffer(), O_WRONLY | O_CREAT, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH);
       if (rc != -1) {
          close(rc);
          unlink(tfn.getBuffer());

